### ISSUE TITLE
No mutating strings rule?

### ISSUE THREAD

**Issue by mockdeep** [rubocop-hq/rubocop#3298]
We are debating whether to enable the `FrozenStringLiteralComment` rule. The problem is that having this magic comment across our codebase is a little invasive. It would be great to just have a rule that doesn't allow common string mutations instead. In our codebase, the only mutation we have is the `<<` operator in a couple of places, so disallowing this as a cop would be great instead of having to add the comment to all of our files.

**Comment by Drenmi** [comment]
This could be a nice complement.

Some mutating methods, e.g. `#<<`, `#[]=`, `#concat`, etc. would only work on literals, since `Array` (and some others) share the same method names.

Others, e.g. `#squeeze!`, `#capitalize!`, `#swapcase!`, etc., which are unique to `String`, would also be possible for variable references.

**Comment by basex** [comment]
I think the `FrozenStringLiteralComment` should accept the option `EnforcedStyle: when_needed` even if ruby is above 2.3. Currently this condition is preventing that:

```ruby
return if style == :when_needed && target_ruby_version < 2.3
```
https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/style/frozen_string_literal_comment.rb

The current auto correct of this COP is not safe with ruby > 2.3 as it adds the comment everywhere without any check.
